libs/ui/src/format_usb_modal.tsx

```diff
@@ -1,29 +1,25 @@
-import React, { useCallback, useContext, useState } from 'react';
-import { assert, throwIllegalValue } from '@votingworks/basics';
-
-import { Button, Modal, Loading, P, Icons, Font } from '@votingworks/ui';
-import {
-  isElectionManagerAuth,
-  isSystemAdministratorAuth,
-} from '@votingworks/utils';
-import { AppContext } from '../contexts/app_context';
-import { formatUsbDrive } from '../api';
-
-export interface FormatUsbModalProps {
-  onClose: () => void;
-}
+import React, { useCallback, useState } from 'react';
+import { assert, Result, throwIllegalValue } from '@votingworks/basics';
+import { UsbDriveStatus } from '@votingworks/usb-drive';
+import { UseMutationResult } from '@tanstack/react-query';
+import { Button } from './button';
+import { Modal } from './modal';
+import { Font, P } from './typography';
+import { Icons } from './icons';
+import { Loading } from './loading';
 
 type FlowState =
   | { stage: 'confirm' }
   | { stage: 'formatting' }
   | { stage: 'done' }
   | { stage: 'error'; message: string };
 
-function FormatUsbFlow({ onClose }: FormatUsbModalProps): JSX.Element {
-  const { usbDriveStatus, auth } = useContext(AppContext);
+function FormatUsbFlow({
+  onClose,
+  usbDriveStatus,
+  formatUsbDriveMutation,
+}: FormatUsbModalProps): JSX.Element {
   assert(usbDriveStatus.status !== 'no_drive');
-  assert(isSystemAdministratorAuth(auth) || isElectionManagerAuth(auth));
```

Comment:
Since this now exists in` libs/ui` it's up to the caller to enforce auth permissions. Also, I believe the election manager auth check is outdated because there is no place in VxAdmin that allows an election manager to format a USB drive.